execution/aggregate/khashaggregate.go

```diff
@@ -293,10 +320,30 @@ func (a *kAggregate) aggregate(t int64, result *[]model.StepVector, k int, sampl
 				}
 			}
 		}
+	case parser.LIMIT_RATIO:
+		for i, sId := range sampleIDs {
+			sampleHeap := a.inputToHeap[sId]
+
+			if addRatioSample(ratio, a.series[sId]) {
+				heap.Push(sampleHeap, &entry{sId: sId, total: samples[i]})
+			}
+		}
+
+		for i, histId := range histogramIDs {
+			sampleHeap := a.inputToHeap[histId]
+
+			if addRatioSample(ratio, a.series[histId]) {
+				heap.Push(sampleHeap, &entry{histId: histId, histogramSample: histograms[i]})
+			}
+		}
 	}
 
 	s := a.vectorPool.GetStepVector(t)
 	for _, sampleHeap := range a.heaps {
+		// for topk and bottomk the heap keeps the lowest value on top, so reverse it.
+		if (a.aggregation == parser.TOPK || a.aggregation == parser.BOTTOMK) && len(sampleHeap.entries) > 1 {
```

Comment:
Very small nit: maybe we can just call `sort.Sort(...)` without checking `len(sampleHeap.entries) > 1`? If there's only 1 or 0 elements, then it will do nothing either way.